visualization/app/codeCharta/util/nodeDecorator.ts

```diff
@@ -19,24 +18,25 @@ const enum EdgeAttributeType {
 
 export class NodeDecorator {
 	static decorateMap(map: CodeMapNode, metricData: MetricData, blacklist: BlacklistItem[]) {
-		const flattened = ignore()
-		const excluded = ignore()
-
-		let hasFlattenedPaths = false
-		let hasExcludedPaths = false
-
 		for (const item of blacklist) {
-			const path = transformPath(item.path)
-
-			if (item.type === BlacklistType.flatten) {
-				hasFlattenedPaths = true
-				flattened.add(path)
-			} else {
-				hasExcludedPaths = true
-				excluded.add(path)
+			for (const { data } of hierarchy(map)) {
+				if (blacklist.length > 0) {
+					if (item.type === BlacklistType.flatten) {
+						data.isFlattened = data.isFlattened ? true : IsNodeExcludedOrFlattened(data, item.path)
+					} else {
+						data.isExcluded = data.isExcluded ? true : IsNodeExcludedOrFlattened(data, item.path) && isLeaf(data)
+					}
+					if (data.type === "Folder") {
```

Comment:
```suggestion
					if (data.type === NodeType.FOLDER") {
```
Do not forget to import the `NodeType` `enum`.

Reply:
I will remove this part because in the next task I noticed it is actually not needed and changes the behaviour of the App abit